src/main/java/org/javarosa/xpath/analysis/XPathAccumulatingAnalyzer.java

```diff
@@ -24,41 +25,58 @@
  */
 public abstract class XPathAccumulatingAnalyzer<T> extends XPathAnalyzer {
 
-    private List<T> accumulatedList = new ArrayList<>();
+    private Collection<T> accumulated;
```

Comment:
I think we rely on the order of this accumulated list to make sure that results are ordered reasonably in traces (IE: results are returned in document / evaluation order).

There are hashsets which retain order as well as having set characteristics, I think it'd be worth mandating by type that this is ordered so that we don't lose that value

Reply:
That hasn't actually changed from the way I had it before; not sure if you realized that and were just making the comment anyway? before this refactor, this was still getting implemented as an unordered set the majority of the time.